tests/tests/fee_tests.cpp

```diff
@@ -213,6 +213,102 @@ BOOST_AUTO_TEST_CASE(asset_claim_fees_test)
+                              const asset& amount_to_fund, const asset_object& fee_asset  )
+        {
+            asset_claim_pool_operation claim_op;
+            claim_op.issuer = issuer;
+            claim_op.asset_id = asset_to_claim;
+            claim_op.amount_to_claim = amount_to_fund;
+
+            signed_transaction tx;
+            tx.operations.push_back( claim_op );
+            db.current_fee_schedule().set_fee( tx.operations.back(), fee_asset.options.core_exchange_rate );
+            set_expiration( db, tx );
+            sign( tx, alice_private_key );
+            PUSH_TX( db, tx );
+
+        };
+
+        const asset_object& core_asset = asset_id_type()(db);
+
+        if( db.head_block_time() <= HARDFORK_CORE_188_TIME )
+        {
+           // can't claim pool before hardfork
+           GRAPHENE_REQUIRE_THROW( claim_pool( alice_id, aliceusd.id, _core(200), core_asset), fc::exception );
+
+           generate_blocks( HARDFORK_CORE_188_TIME );
+           while( db.head_block_time() <= HARDFORK_CORE_188_TIME )
+           {
+              generate_block();
+           }
+        }
+
+        // can't claim pool because it is empty
+        GRAPHENE_REQUIRE_THROW( claim_pool( alice_id, aliceusd.id, _core(1), core_asset), fc::exception );
+
+        // deposit 300 BTS to the fee pool of ALICECOIN asset
+        fund_fee_pool( alice, alicecoin, _core(300).amount );
+
+        // deposit 100 BTS to the fee pool of ALICEUSD asset
+        fund_fee_pool( alice, aliceusd, _core(100).amount );
+
+        BOOST_CHECK( alicecoin.dynamic_asset_data_id(db).fee_pool == _core(300).amount );
+
+        // can't claim pool of an asset that doesn't belong to you
+        GRAPHENE_REQUIRE_THROW( claim_pool( alice_id, bobcoin_id, _core(200), core_asset), fc::exception );
+
+        // can't claim more than is available in the fee pool
+        GRAPHENE_REQUIRE_THROW( claim_pool( alice_id, alicecoin.id, _core(400), core_asset ), fc::exception );
+
+        // can't pay fee in the same asset whose pool is being drained
+        GRAPHENE_REQUIRE_THROW( claim_pool( alice_id, alicecoin.id, _core(200), alicecoin ), fc::exception );
+
+        // can claim BTS back from the fee pool
+        claim_pool( alice_id, alicecoin.id, _core(200), core_asset );
+        BOOST_CHECK( alicecoin.dynamic_asset_data_id(db).fee_pool == _core(100).amount );
+
+        // can pay fee in the asset other than the one whose pool is being drained
+        share_type balance_before_claim = get_balance( alice_id, asset_id_type() );
+        claim_pool( alice_id, alicecoin.id, _core(100), aliceusd );
+        BOOST_CHECK( alicecoin.dynamic_asset_data_id(db).fee_pool == _core(0).amount );
+
+        //check balance after claiming pool
+        share_type current_balance = get_balance( alice_id, asset_id_type() );
+        BOOST_CHECK( balance_before_claim + _core(100).amount == current_balance );
+
+    }
```

Comment:
Best if have a unit test case for `proposal_create_operation`.

Reply:
Erm, not sure how to actually do that .. You are basically asking for a test case for the `hf_*_vistitor`?

Reply:
Yes, something like the indirect test [here](https://github.com/bitshares/bitshares-core/blob/develop/tests/tests/fee_tests.cpp#L1054-L1159) in comparison to the direct test.

Reply:
Would [this](https://github.com/bitshares/bitshares-core/pull/572/commits/c260938efb1f7a6a9a26cd76c9a0f5d4dd87338f) be sufficient to test the visitor?

Reply:
Still need another one test after hard fork. Thanks.